src/init.cpp

```diff
@@ -155,8 +155,24 @@ static fs::path GetPidFile(const ArgsManager& args)
 // shutdown thing.
 //
 
+#if HAVE_SYSTEM
+static void ShutdownNotify(const ArgsManager& args)
+{
+    std::vector<std::thread> threads;
+    for (const auto& cmd : args.GetArgs("-shutdownnotify")) {
+        threads.emplace_back(runCommand, cmd);
```

Comment:
Is it possible to Restrict `runCommand` so that it will use only local resources and network?

Reply:
Do you mean we should restrict all shell commands to only interact with localhost (ie do not allow curl calls to any external hosts)? Apart from performing some sort of validation on the string provided for the option, I am unaware of any technique to enforce this. Do you have any suggestions?

Reply:
As I mentioned earlier, this is unrelated to this pull and might be best discussed in another issue/pull for *all* notify commands. https://github.com/bitcoin/bitcoin/pull/23395#issuecomment-956356904

Reply:
Yes because attackers can't do much if requests only work for localhost. I will ask in few subreddits and see if there are any ways to achieve this.

Reply:
It is related because if something was missed in other options previously we can try to make this secure and follow the same for others.

And I have created an issue with all details.

Reply:
@MarcoFalke agreed, will move the conversation to #23412